src/containers/NoMatch/index.jsx

```diff
@@ -1,54 +1,58 @@
-import { Component } from 'react'
 import PropTypes from 'prop-types'
-import { withTranslation } from 'react-i18next'
+import { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
 import { analytics, ANALYTIC_TYPES } from '../shared/utils'
 import InfoIcon from '../shared/images/info.svg'
 import './nomatch.scss'
 
-class NoMatch extends Component {
-  componentDidMount() {
-    const { t, title, hints } = this.props
+const NoMatch = (props) => {
+  const { t } = useTranslation()
+  const { title, hints, isError, warning } = props
+
+  useEffect(() => {
     document.title = `${t('xrpl_explorer')} | ${t(title)}`
     analytics(ANALYTIC_TYPES.pageview, {
       title: `${title} -- ${hints.join(', ')}`,
       path: '/404',
     })
-  }
+  }, [hints, title, t])
 
-  render() {
-    const { t, title, hints, isError = true } = this.props
-    const notFound = title.includes('not_found')
-    const hintMsg = hints.map((hint) => (
-      <div className="hint" key={hint}>
-        {t(hint)}
-      </div>
-    ))
+  const notFound = title.includes('not_found')
```

Comment:
Should this be documented somewhere so that future people adding `not_found`-like pages know to do this?